File: benchmarks/poly/factorial/factorial.js
function fact(n) {
  if (n < 2) return 1;
  return n * fact(n - 1);
}

function main() {
  const _t0 = process.hrtime.bigint();
  for (let i = 0; i < 10; i++) {
    console.log(fact(i));
  }
  console.error(`bench_ns=${process.hrtime.bigint() - _t0}`);
}

main();

File: benchmarks/poly/factorial/factorial.rs
fn fact(n: i64) -> i64 {
    if n < 2 {
        return 1;
    }
    n * fact(n - 1)
}

fn main() {
    let t0 = std::time::Instant::now();
    for i in 0..10 {
        println!("{}", fact(i));
    }
    eprintln!("bench_ns={}", t0.elapsed().as_nanos());
}
